contentcuration/contentcuration/frontend/shared/vuex/file/actions.js

```diff
@@ -281,3 +281,29 @@ export function getAudioData(context, url) {
       .catch(reject);
   });
 }
+
+export function downloadFile(context, { url, fileName }) {
+  return new Promise((resolve, reject) => {
+    client
+      .get(url, {
```

Comment:
Why shouldn't the link set the source URL directly? Does this blob download cause the browser to buffer it meaning for a large file, it must download the entire file before it actually initiates download? Because that delay could be a confusing experience

Reply:
If using the axios client is to get around the redirect that happens in local dev, I don't think this is worth having if there's a delay due to the blob handling.

Reply:
I guess the redirect that happens in local dev, forcing a display of the file got me exploring the use of the axios client.

Reply:
Is there a way to test this on local dev -- forcing an actual download instead. Maybe using another file type?

Reply:
Noting that we did a similar simplification in Kolibri recently: https://github.com/learningequality/kolibri/pull/12675

I wonder if there's a way that we can have the download <a> directive handle the redirect?

Reply:
I imagine the issue is caused by this:
> download only works for [same-origin URLs](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy), or the blob: and data: schemes.

So, when we redirect to another domain (localhost:9000) it no longer initiates a download and just links to it.

Reply:
I suspect that this may be more trouble than it's worth to fix solely for development - so we may just have to accept that in development the download button will redirect.

Reply:
Yes, it seems to be a same-origin urls issue.